packages/@aws-cdk/aws-events/lib/target.ts

```diff
@@ -53,6 +53,12 @@ export interface RuleTargetConfig {
    */
   readonly ecsParameters?: CfnRule.EcsParametersProperty;
 
+  /**
+   * Parameters used when the rule invokes Amazon API Gateway
+   * @default no parameters set
+   */
+  readonly httpParameters?: CfnRule.HttpParametersProperty;
```

Comment:
each `target` that has custom-ish props has to be added to here.  I am wondering if we should allow any `CfnRule` types to be passed in here? like `readonly extraParameters?: CfnRule;` <--- not sure if we can generify that though.

I would have been able to create a local Level 2 construct for myself without needing to create a PR for this if it were the case... Therefore, I'm not sure if it's a good or a bad thing to make this more generic :)

Reply:
I don't think we have a need to optimize this just yet, but open to ideas if you have any :)